fedmsg/crypto/x509.py

```diff
@@ -113,7 +113,8 @@ def fail(reason):
     for field in ['signature', 'certificate']:
         if field not in message:
             return fail("No %r field found." % field)
-        if not isinstance(message[field], str):
+        if not isinstance(message[field], basestring) and \
```

Comment:
So it is, I did a terribly job rebasing. Then we *do* need this patch, just reworked.